app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java

```diff
@@ -536,4 +548,146 @@ public ActionItem(int[] data) {
             stringId = data[2];
         }
     }
+
+    protected void getAndInsertDate() { getAndInsertDate("",0); }
+
+    protected void getAndInsertDate(String prefix) { getAndInsertDate(prefix, 0); }
+
+    protected void getAndInsertDate(int deltaDays) { getAndInsertDate("", deltaDays); }
+
+    protected void getAndInsertDate(String prefix, int deltaDays) {
+
+        final int[] selection = StringUtils.getSelection(_hlEditor);
+        Editable text = _hlEditor.getText();
+
+        if (_activity != null && _activity instanceof FragmentActivity) {
+
+            DatePickerDialog.OnDateSetListener listener = (view, year, month, day) -> {
+                Calendar calendar = Calendar.getInstance();
+                calendar.set(year, month, day);
+                String date = prefix + DATEF_YYYY_MM_DD.format(calendar.getTime());
+                text.replace(selection[0], selection[1], date);
+            };
+
+            Calendar calendar = Calendar.getInstance();
+
+            // Parse selection for date use if found
+            try {
+                CharSequence selText = text.subSequence(selection[0], selection[1]);
+                Matcher match = SttCommander.PATTERN_IS_DATE.matcher(selText);
+                if (match.find()) calendar.setTime(DATEF_YYYY_MM_DD.parse(selText.toString()));
+            } catch (ParseException e) {
+                // Regex failed?
+                e.printStackTrace();
+            }
+
+            // Add requested offset
+            calendar.add(Calendar.DATE, deltaDays);
+
+            DateTimeFragment dateFragment = new DateTimeFragment()
+                    .setActivity(_activity)
+                    .setDateListener(listener)
+                    .setCalendar(calendar)
+                    .setExtra("Advanced", new DialogInterface.OnClickListener() {
```

Comment:
Added an 'advanced' button which kicks you onto the advanced date and time dialog